### ISSUE TITLE
(Matching Service) Matching logic to match two users

### ISSUE THREAD

**Issue by tanyjnaaman** [CS3219-AY2223S1/cs3219-project-ay2223s1-g47#11]
This should be done after #22.

- [ ] Implement logic of matching by difficulty (separate queues? How might the data be sent from frontend?)
Consider extensibility with future nice-to-haves in mind - private rooms, roles
